core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -953,6 +953,9 @@ private Object readResolve() {
     @Nonnull
     @Restricted(NoExternalUse.class)
     public InstallState getInstallState() {
+        if (installState == null || installState.name() == null) {
+            return InstallState.UNKNOWN;
```

Comment:
Maybe, but there is absolutely nothing that anyone can do about it here. When did it get corrupted? This gives no information why it's in a bad state.